src/tyr/route_serializer_osrm.cc

```diff
@@ -682,13 +648,35 @@ valhalla::baldr::json::MapPtr serializeIncident(const TripLeg_Node_Incident& inc
   if (incident.type()) {
     metadata_json->emplace("incident_type", valhalla::incidentTypeToString(incident.type()));
   }
-  if (!incident.description().empty()) {
+  if (incident.has_description() && !incident.description().empty()) {
     metadata_json->emplace("description", incident.description());
   }
+  if (incident.has_begin_shape_index()) {
+    metadata_json->emplace("geometry_index_start",
+                           static_cast<uint64_t>(incident.begin_shape_index()));
+  }
+  if (incident.has_end_shape_index()) {
+    metadata_json->emplace("geometry_index_end", static_cast<uint64_t>(incident.end_shape_index()));
+  }
 
   return metadata_json;
 }
 
+// Serializes incidents and adds to json-document
+void serializeIncidents(
+    const google::protobuf::RepeatedPtrField<valhalla::TripLeg::Incident>& incidents,
+    json::MapPtr json_leg) {
+  if (incidents.empty()) {
+    return;
+  }
+  auto serialized_incidents = json::array({});
+  for (const auto& incident : incidents) {
+    auto json_incident = serializeIncident(incident);
+    serialized_incidents->emplace_back(json_incident);
+  }
+  json_leg->emplace("incidents", serialized_incidents);
+}
```

Comment:
here we add all the incidents to the leg. deleted the other function that did it for the node above